File: beginner/c/03_is_even/solution.c
#include <stdio.h>

/**
 * Determine if a number is even
 * 
 * @param n The number to check
 * @return 1 if n is even, 0 if n is odd
 * 
 * TODO: Implement this function
 */
int is_even(int n) {
    if (n % 2 ==0){
        return 1;
    } else {
        return 0;
    }
}

int main() {
    // Test your function
    printf("is_even(4) = %d\n", is_even(4));
    printf("is_even(7) = %d\n", is_even(7));
    printf("is_even(0) = %d\n", is_even(0));
    printf("is_even(-2) = %d\n", is_even(-2));
    return 0;
}

File: beginner/java/03_is_even/Solution.java
/**
 * Challenge 03: Is Even
 * TODO: Implement the isEven method
 */
public class Solution {
    
    /**
     * Determine if a number is even
     * 
     * @param n The number to check
     * @return true if n is even, false if n is odd
     */
    public static boolean isEven(int n) {
        return n % 2 == 0;
    }
    
    public static void main(String[] args) {
        // Test your method
        System.out.println("isEven(4) = " + isEven(4));
        System.out.println("isEven(7) = " + isEven(7));
        System.out.println("isEven(0) = " + isEven(0));
        System.out.println("isEven(-2) = " + isEven(-2));
    }
}
